### DISCUSSION TITLE
Bad Request: No such model/language/tier combination found. (Status: 400)

### DISCUSSION THREAD

**Discussion by julien9214** [deepgram/community#930] [General help] [answered]
Hi,

I'm trying to use the PreRecorded Audio Transcription API and I'm having trouble with the following error:
"Bad Request: No such model/language/tier combination found. (Status: 400)"

I've tried to change the parameters (language, model, and smart_format) without success. The audio files to transcribe could be French or English, btw.

Also, I've noticed that even though I change the parameters in my code, the URL doesn't seem to change in the logs (on the Deepgram platform): /v1/listen?language=en-EN&model=nova-2&smart_format=true. For example, if I switch to language="fr-FR", the URL stays as language=en-EN.

Here are the request IDs for the first and last attempts:
First: 2ef174e2-da55-4890-a23b-aa6ef9cb1aab
Last: 11a3f138-976b-4f11-b8fc-5bd191e0cdb5

```
from deepgram import DeepgramClient, PrerecordedOptions

def transcribe_audio_url(audio_url):
    try:
        client = DeepgramClient(Config.DEEPGRAM_API_KEY)
        
        source = {'url': audio_url}
        
        options = PrerecordedOptions(
            model="nova-2",
            language="fr-FR",
            smart_format=True,
        )
        
        response = client.listen.rest.v("1").transcribe_url(source, options)
        url_response = json.loads(response.to_json())
        
        transcript = url_response['results']['channels'][0]['alternatives'][0]['transcript']
        return transcript

    except Exception as e:
        error_details = traceback.format_exc()
        print(f"Detailed error:\n{error_details}")
        raise Exception(f"Transcription failed: {str(e)}")
```

Thanks in advance to anyone that might be able to help!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment, marked as answer]
Hi @Julien9214 

Based on the error:
> "Bad Request: No such model/language/tier combination found. (Status: 400)"

Take a look at the documentation with the model support:
https://developers.deepgram.com/docs/models-languages-overview

For `nova-2`, languages I presume you are trying to use:
French: `fr`, `fr-CA`

There isn't a `fr-FR` for `nova-2`.

**Reply by julien9214** [reply]
Thanks for your answer!

I've just tried to with the following code but I still get the same error :
```
options = PrerecordedOptions(
            model="nova-2",
            language="fr",
            smart_format=True,
        )
```
<br>
And it seems that the URL is still the same, saying that language is `en-EN` even though I sent `fr`.

<img width="786" alt="Capture d’écran 2024-09-22 à 11 48 33" src="https://github.com/user-attachments/assets/6c43d5d8-8bfe-4162-ae59-a625610ba457">
<br><br>
I've also tried to use `en-EN` but I get the same error.

**Comment by davidvonthenen** [comment]
That is very odd considering there isn't anything in the SDK that sets the language to a value other than the one you provide. Can you enable debug and paste the trace here?

https://github.com/deepgram/deepgram-python-sdk/blob/main/examples/speech-to-text/rest/file/main.py#L27

**Reply by julien9214** [reply]
I have no idea why, but everything started working again: the URL gets the `fr` language parameter and the transcription is returned as intended.

Thank you for your help!

**Comment by davidvonthenen** [comment]
Also @Julien9214 

The response object also supports accessing the values in multiple ways. So instead of converting to a JSON object as you are doing in your code:

```
     response = client.listen.rest.v("1").transcribe_url(source, options)
        url_response = json.loads(response.to_json())
        
        transcript = url_response['results']['channels'][0]['alternatives'][0]['transcript']
        return transcript
```

you could do (code:
https://github.com/deepgram/deepgram-python-sdk/blob/main/examples/speech-to-text/rest/url/main.py#L35-L40):

```
     response = client.listen.rest.v("1").transcribe_url(source, options)
        
        transcript = response.results.channels[0].alternatives[0].transcript
        return transcript
```

or even:

```
 response = client.listen.rest.v("1").transcribe_url(source, options)
        
        transcript = response['results']['channels'][0]['alternatives'][0]['transcript']
        return transcript
```

**Reply by julien9214** [reply]
Just made the change. Thanks for the tip !

**Comment by yadavraju** [comment]
Deepgram is not doing great job on language Tamil it skip 90 percent of words

**Comment by AddySrivastava** [comment]
@davidvonthenen - We are trying to use Deepgram Voice AI Nova-3 Multilingual Speech-to-Text (STT)] Streaming marketplace offering via sagemaker endpoint and while the endpoint created successfully, we are running into the following error - 

`2026-07-02T07:31:24.937145677Z WARN request{id=45ed4e7e-9aa0-4abe-bf0e-b2ca3ebc7bbe parent_id=None}:resolve_one_model: impeller::actors::web: Could not find a model that matched the requested model/language/version/tier combination model=general version=latest language=en tier=nova`

We've tried providing multiple options in the model , eg- nova3, general-nova-3, medical-nova3 but none of these options worked, we are running the transcription via following script - https://github.com/deepgram-devs/dg-sagemaker/blob/main/python-stt/stt_wav_stress.py but running into the above error every time.

**Reply by VWang1111** [reply]
**Root cause:** the Nova-3 Multilingual SageMaker listing loads a single model registered under `language=multi`. Your requests are using the default `language=en` (with `model=general`, `tier=nova`/`nova-3`), which doesn't match anything in that listing — hence "Could not find a model that matched the requested model/language/version/tier combination."

**Fix:** set `language=multi` on the request. `model`/`tier` don't need to change — leave those as `general`/`nova-3`.

If you're using the `dg-sagemaker` test script referenced above, that's a `--language multi` flag on `stt_wav_stress.py stream`. If you're using the Deepgram Python SDK directly, set `language="multi"` in `LiveOptions`/`PrerecordedOptions`.

I reproduced your exact error against a Nova-3 Multilingual streaming endpoint with `language=en` and confirmed `language=multi` resolves it — got real transcripts back.
